### PR TITLE
liquid-dsp 1.2.0 (new formula)

### PR DESCRIPTION
**Pull request by muellermartin**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by dunn** [comment]
Failing to build on Mavericks: http://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/4200/version=mavericks/testReport/junit/brew-test-bot/mavericks/install_liquid_dsp/

**Comment by muellermartin** [comment]
Yup, already saw that, but it seems to be a weird inline assembler (?) related error and I'm curious why this only only happens on Mavericks. I guess this could be a compiler issue...

**Comment by dunn** [comment]
Could you report it to the developers? Thanks!

**Comment by muellermartin** [comment]
@BrewTestBot Please test this again.

**Comment by MikeMcQuaid** [comment]
@muellermartin Why did you close/reopen this?

**Comment by muellermartin** [comment]
@MikeMcQuaid Sorry, I accidentaly clicked the wrong button :(

**Comment by muellermartin** [comment]
I just wanted to note that I found a similar issue with OP2/PyOP2#471 which points out that this could be related to AVX instructions. With the last commit I want to check if the build error on Mavericks can be resolved with the suggested fix (adding `-Wa,-q` to the `CFLAGS`) which is also suggested in [this StackOverflow thread](https://stackoverflow.com/questions/9840207/how-to-use-avx-pclmulqdq-on-mac-os-x).

**Comment by muellermartin** [comment]
Build looks good, no errors - even on Mavericks. I'll report that issue upstream. Until then this package could be used.

File: Formula/liquid-dsp.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  homepage "http://liquidsdr.org/"
+  url "https://github.com/jgaeddert/liquid-dsp/archive/v1.2.0.tar.gz"
+  sha256 "783854b63f5e9a9830dcd95b2c1d5c0bef7d10d22a0e27844a354b59283b6de3"
+
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
+  depends_on "fftw" => :recommended
+  # libfec could be an optional dependency, but it's not available, yet
+
+  fails_with :clang do
```
Comment (DomT4) [inline Formula/liquid-dsp.rb]: We've been extremely 👎 on accepting new formulae which won't build against Clang, particularly if there's no upstream work in progress on making it Clang compatible.

File: Formula/liquid-dsp.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+class LiquidDsp < Formula
+  desc "Signal processing (DSP) library for software-defined radios"
+  homepage "http://liquidsdr.org/"
+  url "https://github.com/jgaeddert/liquid-dsp/archive/v1.2.0.tar.gz"
+  sha256 "783854b63f5e9a9830dcd95b2c1d5c0bef7d10d22a0e27844a354b59283b6de3"
+
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
+  depends_on "fftw" => :recommended
+  # libfec could be an optional dependency, but it's not available, yet
```
Comment (DomT4) [inline Formula/liquid-dsp.rb]: Don't need this comment.

Comment (muellermartin) [inline Formula/liquid-dsp.rb]: Removed in commit e559956

File: Formula/liquid-dsp.rb (hunk 1 above)
Comment (muellermartin) [inline Formula/liquid-dsp.rb]: I've found a way to successfully build it with LLVM/Clang instead of GCC with a simple patch which already has an upstream issue (see jgaeddert/liquid-dsp#50). Should I submit another pull request or add it here?

Comment (DomT4) [inline Formula/liquid-dsp.rb]: Submitting a PR & then using that PR in `patch do` format if it applies cleanly here would be terrific.

Comment (muellermartin) [inline Formula/liquid-dsp.rb]: I've already submitted a patch upstream (jgaeddert/liquid-dsp#51) Do I understand it correctly that I can add it here in this PR? Is an embedded patch (`patch :DATA` + `__END__`) O.K.?

Comment (DomT4) [inline Formula/liquid-dsp.rb]: If it will apply cleanly, we'd prefer the:

```ruby
patch do
  url "https://github.com/jgaeddert/liquid-dsp/pull/51.patch"
  sha256 ""
end
```
Format.

Comment (muellermartin) [inline Formula/liquid-dsp.rb]: Ah, didn't know you also can get a "raw" patch file out of a PR hosted on GitHub. Neat and thanks for this tip!

Comment (muellermartin) [inline Formula/liquid-dsp.rb]: Added commit with patch.

Comment (DomT4) [inline Formula/liquid-dsp.rb]: Aye, you can get diffs if preferred by simply switching it to `diff` as well. Very handy feature.

File: Formula/liquid-dsp.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  # Fix configure checks to be able to compile with LLVM/Clang
+  patch do
+    url "https://github.com/jgaeddert/liquid-dsp/pull/51.patch"
+    sha256 "54dc680ba12c7ed89f667e99b84669770c8b56100df47e4b9ba2f793401c4a96"
+  end
+
+  def install
+    system "./reconf"
+    system "./configure", "--prefix=#{prefix}"
+    # Note: "make install" in one step does fail
```
Comment (MikeMcQuaid) [inline Formula/liquid-dsp.rb]: Could you report this issue upstream? Thanks!

File: Formula/liquid-dsp.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  homepage "http://liquidsdr.org/"
+  url "https://github.com/jgaeddert/liquid-dsp/archive/v1.2.0.tar.gz"
+  sha256 "783854b63f5e9a9830dcd95b2c1d5c0bef7d10d22a0e27844a354b59283b6de3"
+  head "https://github.com/jgaeddert/liquid-dsp.git"
+
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
+  depends_on "fftw" => :recommended
+
+  # Fix configure checks to be able to compile with LLVM/Clang
```
Comment (MikeMcQuaid) [inline Formula/liquid-dsp.rb]: Some conflicts that need resolved upstream on this. Would like to see upstream accept this and at least implicitly state they are wanting to support Clang before merging this.

File: Formula/liquid-dsp.rb (hunk 3 above)
Comment (muellermartin) [inline Formula/liquid-dsp.rb]: I'll do that and report back as soon as it's done!

File: Formula/liquid-dsp.rb (hunk 4 above)
Comment (muellermartin) [inline Formula/liquid-dsp.rb]: I heard a lot issues are resolved in master and the developer is about to publish a new release because the tarballs are "woefully out of date" (see https://github.com/jgaeddert/liquid-dsp/issues/49#issuecomment-231588827).

Comment (dunn) [inline Formula/liquid-dsp.rb]: Good news!

File: Formula/liquid-dsp.rb (hunk 3 above)
Comment (muellermartin) [inline Formula/liquid-dsp.rb]: I've now reported this upstream as jgaeddert/liquid-dsp#53 and included a simple PR. Should I add a "patch" with inreplace in the formula to change the relevant file in the Makefile (makefile.in) or just add/replace the comment to mention the upstream bug?

Comment (MikeMcQuaid) [inline Formula/liquid-dsp.rb]: Just add the comment, thanks 👍

**Comment by muellermartin** [comment]
I've now added a comment on another upstream issue as suggested. The reported upstream issues (jgaeddert/liquid-dsp#48 and jgaeddert/liquid-dsp#52) do have patches/workarounds until they get resolved and released which seems to take time due to stalling maintenance, but I've heard that this is beeing worked on.
If you guys are satisfied with the current state of the formula, feel free to merge it.

**Comment by dunn** [comment]
I might wait another day or two for upstream to reply to your tickets (thanks for filing them!)

**Comment by MikeMcQuaid** [comment]
Closing until we get a new stable release with these patches included.
